store/PoolsSlice/index.tsx

```diff
@@ -9,9 +9,17 @@ export const createPoolsSlice: StateSlice<IPoolsSlice> = (set, get) => ({
     setPoolLists: (network, lists) => {
         set((state) => void (state.poolLists[network] = lists));
     },
-    importPool: (network, pool) => {
+    importPool: (network, poolAddress) => {
         if (get().poolLists[network]) {
-            set((state) => void state.poolLists[network]?.Imported.pools.push({ address: pool }));
+            set((state) => void state.poolLists[network]?.Imported.pools.push({ address: poolAddress }));
+        }
+    },
+    removePool: (network, poolAddress) => {
+        if (get().poolLists[network]) {
+            const poolIndex = get().poolLists[network]?.Imported.pools.findIndex((v) => v.address === poolAddress);
+            if (typeof poolIndex === 'number' && poolIndex >= 0) {
+                set((state) => void state.poolLists[network]?.Imported.pools.splice(poolIndex, 1));
+            }
```

Comment:
up to you but I think its probs a bit cleaner to use `Array.filter` with `(pool) => pool.address !== poolAddress`

Reply:
It accepted `Array.filter` in this form:

![image](https://user-images.githubusercontent.com/19278287/179447960-da328262-4273-4d52-9034-129f844fabc8.png)

but doesn't make any change to the store itself. Will stick to the previous implementation for now I guess